src/phoenix/experimental/evals/templates/template.py

```diff
@@ -16,81 +14,101 @@
-                f"{self._start_delim} and {end_count} instances of {self._end_delim}. "
-                "They must be equal to be correctly paired."
-            )
+    def _parse_variables(self, text: str) -> List[str]:
+        pattern = re.escape(self._start_delim) + "(.*?)" + re.escape(self._end_delim)
+        variables = re.findall(pattern, text)
+        return variables
+
 
-    def _parse_variables(self) -> List[str]:
-        variables = []
-        formatter = Formatter()
-
-        text = self.text
-
-        # Example of this could be a template like: My name is ::name::
-        if self._start_delim == self._end_delim:
-            delim_length = len(self._start_delim)
-            delim_count = text.count(self._start_delim)
-            while delim_count > 0:
-                left_index = text.find(self._start_delim)
-                right_index = text[left_index + delim_length :].find(self._start_delim)
-                text = (
-                    text[0:left_index]
-                    + DEFAULT_START_DELIM
-                    + text[left_index + delim_length : left_index + delim_length + right_index]
-                    + DEFAULT_END_DELIM
-                    + text[left_index + 2 * delim_length + right_index :]
-                )
-                delim_count = text.count(self._start_delim)
+class ClassificationTemplate(PromptTemplate):
+    def __init__(
+        self,
+        rails: List[str],
+        base_template: str,
+        explanation_template: Optional[str] = None,
+        delimiters: Tuple[str, str] = (DEFAULT_START_DELIM, DEFAULT_END_DELIM),
+    ):
+        self.base_template = base_template
+        self.explanation_template = explanation_template
+        self._start_delim, self._end_delim = delimiters
+        self.variables: List[str] = []
+        for text in [base_template, explanation_template]:
+            if text is not None:
+                self.variables += self._parse_variables(text)
+
+    def prompt(self, options: Optional[PromptOptions] = None) -> str:
+        if options is None:
+            return self.base_template
+
+        if options.provide_explanation and self.explanation_template:
+            return self.explanation_template
         else:
-            if self._start_delim != "{":
-                text = text.replace(self._start_delim, DEFAULT_START_DELIM)
-            if self._end_delim != "{":
-                text = text.replace(self._end_delim, DEFAULT_END_DELIM)
+            return self.base_template
 
-        for _, variable_name, _, _ in formatter.parse(text):
-            if variable_name:
-                variables.append(variable_name)
+    def parse_label(self, raw_string: str) -> str:
+        label_delimiter = r"\W*label\W*"
+        parts = re.split(label_delimiter, raw_string, maxsplit=1, flags=re.IGNORECASE)
+        if len(parts) == 2:
+            return parts[1]
+        return NOT_PARSABLE
```

Comment:
this is specific to the output parsing of the explanation so I probably wouldn't leave this generic. Unfortunately we kinda need a parser with and without explanations. We can fall back to this - e.g. make this parser be the default for explanation templates but I wouldn't bake it into the class since it's going to be wrong for some cases.